### PR TITLE
Add support for log rotation parameter and validation

### PR DESCRIPTION
**Pull request by chenwany**
### Description of changes
Add the following configuration to cluster configuration:
```
Monitoring:
    CloudWatch:
      Enabled: false
    Rotation:
      Enabled: true
```
* Add validator to print a warning message when Cloudwatch Enabled is set to false and Rotation is set to true: "Cloudwatch Logs is disabled but Log Rotation is enabled. Logs will be rotated and removed from the cluster once they reach a certain size. If you want to keep logs locally within the cluster, please set `Monitoring / Logs / Rotation / Enabled` to false."

### Tests
* Unit tests

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/tests/pcluster/validators/test_monitoring_validators.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@pytest.mark.parametrize(
+    "logs, expected_message",
+    [
+        (Logs(cloud_watch=CloudWatchLogs(enabled=True), rotation=LogRotation(enabled=True)), None),
+        (Logs(cloud_watch=CloudWatchLogs(enabled=False), rotation=LogRotation(enabled=False)), None),
+        (Logs(rotation=LogRotation(enabled=False)), None),
+        (Logs(rotation=LogRotation(enabled=True)), None),
+        (
+            Logs(cloud_watch=CloudWatchLogs(enabled=False), rotation=LogRotation(enabled=True)),
+            "Cloudwatch Logs is disabled but Log Rotation is enabled. Logs will be rotated and removed from the "
```
Comment (davprat) [inline cli/tests/pcluster/validators/test_monitoring_validators.py]: Nitpick: Would it read better to say `CloudWatch Logging is...`

**Review by davprat — DISMISSED** [review]
Left a non-blocking nitpick.

File: cli/tests/pcluster/validators/test_monitoring_validators.py (hunk 1 above)
Comment (chenwany) [reply cli/tests/pcluster/validators/test_monitoring_validators.py]: Good idea, changed

File: cli/src/pcluster/validators/monitoring_validators.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from pcluster.validators.common import FailureLevel, Validator
+
+
+class LogRotationValidator(Validator):
+    """Security groups validator."""
+
+    def _validate(self, log):
+        if not log.cloud_watch.enabled and log.rotation.enabled:
+            self._add_failure(
+                "Cloudwatch Logging is disabled but Log Rotation is enabled. Logs will be rotated and removed from the "
```
Comment (davprat) [inline cli/src/pcluster/validators/monitoring_validators.py:20]: Nitpick [Non-Blocking]: I think it is supposed to be written `CloudWatch`? nbd though.

**Review by davprat — DISMISSED** [review]

File: cli/src/pcluster/constants.py [hunk 3]
```diff
@@ -111,6 +111,8 @@
 CW_DASHBOARD_ENABLED_DEFAULT = True
 CW_LOGS_ENABLED_DEFAULT = True
 CW_LOGS_RETENTION_DAYS_DEFAULT = 180
+CW_LOGS_ROTATION_ENABLED_DEFAULT = True
+CW_LOGS_RETENTION_DAYS_DEFAULT = 14
```
Comment (davprat) [inline cli/src/pcluster/constants.py]: I think this line needs to be removed?

**Review by davprat — COMMENTED** [review]
Looks the merge left an extra line in.

File: cli/src/pcluster/validators/monitoring_validators.py (hunk 2 above)
Comment (chenwany) [reply cli/src/pcluster/validators/monitoring_validators.py:20]: Good idea, changed

File: cli/src/pcluster/constants.py (hunk 3 above)
Comment (chenwany) [reply cli/src/pcluster/constants.py]: Good catch, it is something wrong I did when rebasing conflict, fixed

**Review by davprat — APPROVED** [review]
lgtm

**Pull request merged**